chain/index/ddls.go

```diff
@@ -0,0 +1,82 @@
+package index
+
+import "database/sql"
+
+const DefaultDbFilename = "chainindex.db"
+
+var ddls = []string{
+	`CREATE TABLE IF NOT EXISTS tipset_message (
```

Comment:
I would prefer to have a separate `chain` table with:

- `tipset_id` (primary key, integer)
- `tipset_key`
- `tipset_cid`
- `height`
- `reverted`
- `finalized`

This will save us quite a bit of disk space and make it easier to lookup tipsets, etc.

Then a separate `tipset_message` table:

- `tipset_id`
- `message_id`
- `message_cid`
- `message_index`

We could further normalize it by splitting out a separate `messages` table for just `message_cid` entries, but, IMO, that's not worth it.

Reply:
@Stebalien 

While this may come with some space saving, this slows down read performance for pretty much any query because all of them then need to do one additional join for the tipset <> message table. We're really aiming for good read performance here.

If we use the schema you are suggesting here, 

- reading a message for a tipset will need one join
- reading an event for a tipset will need two joins
- reading an event entry for a tipset will need **_three_** joins.

The primary goal here is read performance because that is what subgraph providers need. The Index itself is a **_small fraction_** of the total chain state and the space savings aren't going to move the needle at all for RPC providers but will slow down read performance for each query.

Reply:
Last I checked, the index grew to be _quite_ large. Remember, this affects people with the splitstore as well. I guess it'll be better if we have proper GC, but have we measured the performance impact of that join? My guess is that the performance will be just fine.

More importantly, we want a chain table anyways.

Reply:
@Stebalien 

- The Index can be large because the current Indexing system does NOT have GC. We're shipping Index GC with this new Indexer that will be enabled by RPC providers with splitstore enabled.

- On an archival node, state size is 15TB and Index size is a few hundred GBs so there is no meaningful space saving here.

- This IS the chain table. There is an Index on the `height` field that makes it easy and quick to do "give me canonical tipset at height X" kinda queries.  Also, messages are nullable to handle empty tipsets and tipsets for which the chain state does NOT have messages. 

While having a separate table for "tipset" and "messages" feels more aesthetically pleasing and maps better to the "blockchain" view -> it penalises **_every_** read query with an additional join without giving us a meaningful gain anywhere.

Reply:
What if I have a block with no messages? That's totally valid and I won't have a way to, e.g., map a tipset cid to a tipset key without a separate chain table.

Reply:
@Stebalien The message field here is nullable. We index empty tipset as well in this Index.

Reply:
Message field of what? I'm saying: let's assume there's a completely empty tipset with blocks but no messages, how would I query for a tipset key given a tipset CID. If there _are_ no `tipset_message` rows (because there are no messages, there's nothing to query.

Reply:
@Stebalien The `tipset_message` table is designed to handle this scenario. Please see the NULLABLE fields in the DDL. Let's discuss this sync today.

Reply:
Discussed offline with Steb. He agrees that this works for our use case and can be used to extract whatever chain info we need(i.e. we can extract tipset info from this as well).

**_However_**, he wants us to know this is not the cleanest solution and a normalised model will be cleaner.

But, he will not block us on this.